webpack/components/Bookmark/index.js

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector, shallowEqual } from 'react-redux';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { STATUS } from 'foremanReact/constants';
+import { Dropdown, DropdownItem, DropdownToggle, DropdownSeparator } from '@patternfly/react-core';
+import { CaretDownIcon } from '@patternfly/react-icons';
+import { getBookmarks } from './BookmarkActions';
+import { selectBookmarks, selectBookmarkStatus } from './BookmarkSelectors';
+import './Bookmark.scss';
+import AddBookmarkModal from './AddBookmarkModal';
+
+const Bookmark = ({ selectItem, selectedItem, controller = '' }) => {
+  const dispatch = useDispatch();
+  const [dropdownOpen, setDropdownOpen] = useState(false);
+  const [modalOpen, setModalOpen] = useState(false);
+  const { results = [] } =
+    useSelector(state => selectBookmarks(state, controller), shallowEqual);
+  const status =
+    useSelector(state => selectBookmarkStatus(state, controller), shallowEqual);
+
+  useEffect(() => {
+    dispatch(getBookmarks(controller));
+  }, [controller, dispatch]);
+
+  if (status !== STATUS.RESOLVED) return null;
+
+  const setSelectItem = (query) => {
+    if (selectedItem !== query) {
+      selectItem(query);
+    }
+    setDropdownOpen(false);
+  };
+
+  const dropDownItems = [
+    ...results.map(({ name, id, query }) => (
```

Comment:
Is the spread needed? Why not just
```js
const dropdownItems = results.map(...
```

Reply:
I append the add bookmark at the end, its not just the map in this case.

Reply:
Ahhhh I see it now, I had missed that! :+1: